config_src/drivers/FMS_cap/ocean_model_MOM.F90

```diff
@@ -1194,4 +1220,174 @@ subroutine ocean_model_get_UV_surf(OS, Ocean, name, array2D, isc, jsc)
+  endif
+
+  if (associated(IOB%sw_flux_nir_dif)) then
+    allocate(IOB_h%sw_flux_nir_dif (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%sw_flux_nir_dif = 0.0
+    call homogenize_field(IOB%sw_flux_nir_dif,IOB_h%sw_flux_nir_dif, G, index_bounds)
+  endif
+
+  if (associated(IOB%lprec)) then
+    allocate(IOB_h%lprec (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%lprec = 0.0
+    call homogenize_field(IOB%lprec,IOB_h%lprec, G, index_bounds)
+  endif
+
+  if (associated(IOB%fprec)) then
+    allocate(IOB_h%fprec (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%fprec = 0.0
+    call homogenize_field(IOB%fprec,IOB_h%fprec, G, index_bounds)
+  endif
+
+  if (associated(IOB%runoff)) then
+    allocate(IOB_h%runoff (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%runoff = 0.0
+    call homogenize_field(IOB%runoff,IOB_h%runoff, G, index_bounds)
+  endif
+
+  if (associated(IOB%calving)) then
+    allocate(IOB_h%calving (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%calving = 0.0
+    call homogenize_field(IOB%calving,IOB_h%calving, G, index_bounds)
+  endif
+
+  if (associated(IOB%stress_mag)) then
+    allocate(IOB_h%stress_mag (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%stress_mag = 0.0
+    call homogenize_field(IOB%stress_mag,IOB_h%stress_mag, G, index_bounds)
+  endif
+
+  if (associated(IOB%ustar_berg)) then
+    allocate(IOB_h%ustar_berg (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%ustar_berg = 0.0
+    call homogenize_field(IOB%ustar_berg,IOB_h%ustar_berg, G, index_bounds)
+  endif
+
+  if (associated(IOB%area_berg)) then
+    allocate(IOB_h%area_berg (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%area_berg = 0.0
+    call homogenize_field(IOB%area_berg,IOB_h%area_berg, G, index_bounds)
+  endif
+
+  if (associated(IOB%mass_berg)) then
+    allocate(IOB_h%mass_berg (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%mass_berg = 0.0
+    call homogenize_field(IOB%mass_berg,IOB_h%mass_berg, G, index_bounds)
+  endif
+
+  if (associated(IOB%runoff_hflx)) then
+    allocate(IOB_h%runoff_hflx (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%runoff_hflx = 0.0
+    call homogenize_field(IOB%runoff_hflx,IOB_h%runoff_hflx, G, index_bounds)
+  endif
+
+  if (associated(IOB%calving_hflx)) then
+    allocate(IOB_h%calving_hflx (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%calving_hflx = 0.0
+    call homogenize_field(IOB%calving_hflx,IOB_h%calving_hflx, G, index_bounds)
+  endif
+
+  if (associated(IOB%p)) then
+    allocate(IOB_h%p (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%p = 0.0
+    call homogenize_field(IOB%p,IOB_h%p, G, index_bounds)
+  endif
+
+  if (associated(IOB%mi)) then
+    allocate(IOB_h%mi (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%mi = 0.0
+    call homogenize_field(IOB%mi,IOB_h%mi, G, index_bounds)
+  endif
+
+  if (associated(IOB%ice_rigidity)) then
+    allocate(IOB_h%ice_rigidity (isc_bnd:iec_bnd,jsc_bnd:jec_bnd) ) ; IOB_h%ice_rigidity = 0.0
+    call homogenize_field(IOB%ice_rigidity,IOB_h%ice_rigidity, G, index_bounds)
+  endif
+
```

Comment:
@breichl , what about other IOBs? When I run this code change with "generic tracers" turned on the model crashes with 
```
FATAL: extract_coupler_values: The requested boundary condition is not associated.

Image              PC                Routine            Line        Source             
MOM6SIS2           00000000026A4A77  Unknown               Unknown  Unknown
MOM6SIS2           0000000001A421EC  mpp_mod_mp_mpp_er          68  mpp_util_mpi.inc
MOM6SIS2           00000000017767FC  fms_coupler_util_          53  FMS_coupler_util.F90
MOM6SIS2           00000000004E20B6  g_tracer_utils_mp        1429  generic_tracer_utils.F90
MOM6SIS2           00000000011995D0  ocean_model_mod_m         551  ocean_model_MOM.F90
MOM6SIS2           00000000013E3A6F  MAIN__                   1041  coupler_main.F90

```

Reply:
I think that this issue that Niki is encountering suggests that it might be worth revisiting whether it would be simpler to homogenize the fields in the forcing and mech_forcing types, even if we had to add the option to recalculate the dependent fields (e.g., ustar) based on the primary fields (like taux and tauy).  We should not be changing the intent on the various types.